src/CMakeLists.txt

```
@@ -223,9 +226,31 @@ endif()
 
 target_link_libraries(parthenon PUBLIC Kokkos::kokkos)
 
-target_include_directories(parthenon PUBLIC
-  ${CMAKE_CURRENT_SOURCE_DIR}
-  ${CMAKE_CURRENT_BINARY_DIR}/generated
-)
-
 lint_target(parthenon)
+
+target_include_directories(parthenon PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/generated>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+  )
+
+install(TARGETS parthenon EXPORT parthenonTargets
+  INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/parthenon"
+  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    )
+
+# Maintain directory structure in installed include files
+install(DIRECTORY . DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} FILES_MATCHING PATTERN "*.hpp")
+
+# Install generated config header file
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/generated/config.hpp
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

```suggestion
  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/parthenon)
```

Having a `config.hpp` in the general include dir, e.g. `/usr/local/include` is just a no-go.

---

I think this is a more general problem with how the headers are structured right now. For instance, right now we also install `defs.hpp` and `globals.hpp` to `${CMAKE_INSTALL_INCLUDEDIR}`. There was already some discussion earlier about reorganizing headers into a set of public headers to install and a set of private headers to not install, but I think that work will be deferred for a while, so I think the easiest safe thing to do now is just install all headers including `config.hpp` into a `parthenon` subdirectory. Updating the cmake target makes this painless (i.e. no change to `#include <parthenon/parthenon/package.hpp>` in app files). I've gone ahead and done this, thanks for bringing this up.